modules/core/src/main/java/org/apache/ignite/internal/processors/failure/FailureProcessor.java

```diff
@@ -65,7 +78,22 @@
     public FailureProcessor(GridKernalContext ctx) {
         super(ctx);
 
-        this.ignite = ctx.grid();
+        ignite = ctx.grid();
+
+        if (igniteDumpThreadsOnFailure) {
```

Comment:
Any chance we want to also consinder `systemWorkerBlockedTimeout`? checkpoint read lock timeout is calculated like that: take the systemWorkerBlocked if set, else take failure detection timeout.

Reply:
We don't consider any other timeouts.